scipy/sparse/lil.py

```diff
@@ -451,11 +451,18 @@ def tolil(self, copy=False):
 
     def tocsr(self, copy=False):
         lst = [len(x) for x in self.rows]
-        idx_dtype = get_index_dtype(maxval=max(self.shape[1], sum(lst)))
-
-        indptr = np.cumsum([0] + lst, dtype=idx_dtype)
-        indices = np.array([x for y in self.rows for x in y], dtype=idx_dtype)
-        data = np.array([x for y in self.data for x in y], dtype=self.dtype)
+        indptr_dtype = get_index_dtype(maxval=sum(lst))
```

Comment:
As far as I'm aware, most of scipy.sparse assumes that `indptr` and `indices` have the same dtype.

Reply:
Is this used explicitly somewhere? I have used different dtypes without problems and this could save some memory.

Reply:
Stuff gets upcast on every call to any sparsetools c++ function. There's also the chance that some code somewhere breaks with different dtypes.

Reply:
Thank you for the explanation. I have changed it to use the same dtype for both arrays.